File: backend/models/challenge.py
from typing import Literal

from pydantic import BaseModel, Field


class ChatMessage(BaseModel):
    role: Literal["user", "assistant"]  # Block "system" role injection
    content: str = Field(max_length=5000)


class DesignChallengeRequest(BaseModel):
    topic_id: str = Field(max_length=50)
    challenge_id: str = Field(max_length=50)
    messages: list[ChatMessage] = Field(max_length=50)  # max 50 messages in history


class DesignChallengeResponse(BaseModel):
    reply: str
    is_complete: bool = False
    score: float | None = None
    feedback: str | None = None


class AskRequest(BaseModel):
    question: str = Field(max_length=2000)
    context_topic: str | None = Field(default=None, max_length=100)


class AskResponse(BaseModel):
    answer: str

File: frontend/src/types/challenge.ts
export interface ChatMessage {
  role: "user" | "assistant" | "system";
  content: string;
}

export interface DesignChallengeRequest {
  topic_id: string;
  challenge_id: string;
  messages: ChatMessage[];
}

export interface DesignChallengeResponse {
  reply: string;
  is_complete: boolean;
  score: number | null;
  feedback: string | null;
}

export interface AskRequest {
  question: string;
  context_topic?: string;
}

export interface AskResponse {
  answer: string;
}
